include/SFML/Graphics/Text.hpp

```diff
@@ -436,19 +550,31 @@ class SFML_GRAPHICS_API Text : public Drawable, public Transformable
     void ensureGeometryUpdate() const;
 
+    ////////////////////////////////////////////////////////////
+    /// \brief Update underline fill/outline color
+    ///
+    /// If there is no need to update geometry, update colors only.
+    ///
+    ////////////////////////////////////////////////////////////
+    void updateUnderline(VertexArray& arr, const Color& color) const;
+
     ////////////////////////////////////////////////////////////
     // Member data
     ////////////////////////////////////////////////////////////
-    String                m_string;                                    //!< String to display
-    const Font*           m_font{};                                    //!< Font used to display the string
-    unsigned int          m_characterSize{30};                         //!< Base size of characters, in pixels
-    float                 m_letterSpacingFactor{1.f};                  //!< Spacing factor between letters
-    float                 m_lineSpacingFactor{1.f};                    //!< Spacing factor between lines
-    std::uint32_t         m_style{Regular};                            //!< Text style (see Style enum)
-    Color                 m_fillColor{Color::White};                   //!< Text fill color
-    Color                 m_outlineColor{Color::Black};                //!< Text outline color
-    float                 m_outlineThickness{0.f};                     //!< Thickness of the text's outline
-    mutable VertexArray   m_vertices{PrimitiveType::Triangles};        //!< Vertex array containing the fill geometry
+    String                     m_string;                              //!< String to display
+    const Font*                m_font{};                              //!< Font used to display the string
+    unsigned int               m_characterSize{30};                   //!< Base size of characters, in pixels
+    float                      m_letterSpacingFactor{1.f};            //!< Spacing factor between letters
+    float                      m_lineSpacingFactor{1.f};              //!< Spacing factor between lines
+    std::uint32_t              m_style{Regular};                      //!< Text style (see Style enum)
+    Color                      m_underlineFillColor{Color::White};    //!< Text underline/strikethrough fill color
+    Color                      m_underlineOutlineColor{Color::Black}; //!< Text underline/strikethrough outline color
+    bool                       m_multiFillColor{0};          //!< Should the text have more than a single fill color
+    bool                       m_multiOutlineColor{0};       //!< Should the text have more than a single outline color
+    float                      m_outlineThickness{0.f};      //!< Thickness of the text's outline
+    mutable std::vector<Color> m_fillColor{Color::White};    //!< Text fill color(s)
+    mutable std::vector<Color> m_outlineColor{Color::Black}; //!< Text outline color(s)
```

Comment:
```suggestion
    mutable std::vector<Color> m_fillColors{Color::White};    //!< Text fill color(s)
    mutable std::vector<Color> m_outlineColors{Color::Black}; //!< Text outline color(s)
```
Can we pluralize these member variable names so it's more clear that they're now containers?